src/NServiceBus.Core/Diagnostics/TracingExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace NServiceBus;
+
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Pipeline;
+
+static class TracingExtensions
+{
+    public static Task Invoke<TContext>(this IPipeline<TContext> pipeline, TContext context, Activity activity) where TContext : IBehaviorContext
+    {
+        return activity == null ? pipeline.Invoke(context) : TracePipelineStatus();
+
+        async Task TracePipelineStatus()
+        {
+#pragma warning disable PS0019 // When catching System.Exception, cancellation needs to be properly accounted for
+            try
+            {
+                await pipeline.Invoke(context).ConfigureAwait(false);
+                activity.SetStatus(ActivityStatusCode.Ok);
+            }
+            catch (Exception ex)
+            {
+                // TODO: Add an explicit tag for operation canceled
```

Comment:
```suggestion
```

not sure what that would be useful for? We already set the exception details which make it clear that it's an operation cancelled and that doesn't change the stat of the span or does it?